src/viam/ur/module/ur_arm.cpp

```diff
@@ -520,7 +528,12 @@ void URArm::move(std::vector<Eigen::VectorXd> waypoints, std::chrono::millisecon
     const Eigen::VectorXd curr_waypoint_deg =
         Eigen::VectorXd::Map(curr_joint_pos.data(), boost::numeric_cast<Eigen::Index>(curr_joint_pos.size()));
     const Eigen::VectorXd curr_waypoint_rad = curr_waypoint_deg * (M_PI / 180.0);
-    waypoints.insert(waypoints.begin(), curr_waypoint_rad);
+    if (!curr_waypoint_rad.isApprox(waypoints.front(), 0.0001)) {
```

Comment:
- Is it a precondition for this function that `waypoints` be non-empty? It doesn't seem to be enforced.
- Constant for epsilon, should be same as one above I think.
- I think using the `else if` makes it harder to think about. If you make it a standalone conditional it reads better to me:
```
if (!curr_waypoint ... ) {
    waypoints.insert(...);
}

if (waypoints.size() == 1) {
    VIAM_SDK_LOG << ...
    return
}
```

Reply:
yesish, `move_to_joint_positions` and `move_through_joint_positions` should actually validate that  waypoints are non empty, but I don't know if a viam-server can send the module an empty set of waypoints